lib/widgets/movies_repository.dart

```diff
@@ -0,0 +1,57 @@
+import 'dart:convert';
+import 'package:flutter/services.dart';
+
+import '../models/genres.dart';
+import '../models/movie.dart';
+
+class MoviesRepository {
+  static const String genresJsonPath = 'assets/data/genres.json';
+  static const String moviesJsonPath = 'assets/data/movies.json';
+
+  static const String imagesLink = 'https://image.tmdb.org/t/p/w500';
+
+  List<Movie> movies = [];
+  List<Genre> allGenres = [];
+
+  Future<List<Movie>> loadMovies() async {
+    String genreData = await rootBundle.loadString(
+      genresJsonPath,
+    );
+    String movieData = await rootBundle.loadString(
+      moviesJsonPath,
+    );
+
+    List<dynamic> genresFromData = jsonDecode(genreData)['genres'];
+    for (var genre in genresFromData) {
+      allGenres.add(Genre.fromJson(
+        genre,
+      ));
+    }
```

Comment:
Minor improvement, only if you like this approach:
`allGenres = genresFromData.map((genre) => Genre.fromJson(genre));`

Reply:
Right! Done